src/imputors/knn.jl

```diff
@@ -34,40 +34,50 @@ function impute!(data::AbstractMatrix{Union{T, Missing}}, imp::KNN; dims=nothing
     X = d == 1 ? data : transpose(data)
 
     # Get mask array first
-    mmask = ismissing.(X)
+    Xmask = ismissing.(X)
 
-    # fill missing value as mean value
+    # Fill missing value as mean value
     impute!(X, Substitute(); dims=1)
 
     # Disallow `missings` for NearestNeighbors
     X = disallowmissing(X)
 
+    # Our search points are just observations containing `missing`s
+    points = X[:, vec(reduce(|, Xmask; dims=1))]
```

Comment:
```suggestion
    points = X[:, vec(any(Xmask; dims=1))]
```

I wonder if this could be a view
```suggestion
    points = @view X[:, vec(any(Xmask; dims=1))]
```
and if so what the effect that would have on performance.

Reply:
I tried adding `@view` and `@views` calls to all the indexing/slicing calls and it doesn't seem to make much of a difference.

```
julia> @benchmark Impute.knn($X; k=4, dims=:rows)
BenchmarkTools.Trial:
  memory estimate:  33.52 MiB
  allocs estimate:  686426
  --------------
  minimum time:     58.834 ms (0.00% GC)
  median time:      65.528 ms (6.23% GC)
  mean time:        68.600 ms (5.33% GC)
  maximum time:     98.741 ms (13.55% GC)
  --------------
  samples:          73
  evals/sample:     1
```

I guess that makes sense because all of our slices should be a fraction of the size of the original array under normal circumstances (e.g., < 25% missing data, k < 10). My guess is that the overhead of the view outweighs the benefits for relatively small subarrays? I'm also not sure what operations the `NearestNeighbors.knn` call is doing, but those might be less performance for subarrays. I could try benchmarking more extreme cases with more missing values or considering more neighbors, but I don't know how useful that'll be?

Reply:
I think the big performance gain with this PR is with reducing the number of search `points` under normal circumstances.